### PR TITLE
Resolve all cookstyle warnings

### PR DESCRIPTION
**Pull request by tas50**
There was some very funky stuff going on in this cookbook and a lot of code that could be removed. This resolves all the current Cookstyle warnings with cookstyle on the master branch.

### CODE REVIEW

**Comment by michaelklishin** [comment]
Thank you. Any chance you can also submit a version of this for `v5.x`?

File: resources/user.rb [hunk 1]
```diff
@@ -17,8 +17,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
-actions :add, :delete, :set_permissions, :clear_permissions, :set_tags, :clear_tags, :change_password
```
Comment (michaelklishin) [inline resources/user.rb]: This line's removal leads to Kitchen failures:

```
================================================================================
Recipe Compile Error in /opt/kitchen/cache/cookbooks/rabbitmq/recipes/users.rb
================================================================================

Chef::Exceptions::ValidationFailed
----------------------------------
Option action must be equal to one of: :nothing, :add!  You passed :set_tags.

Cookbook Trace:
---------------
  /opt/kitchen/cache/cookbooks/rabbitmq/recipes/users.rb:38:in `block (2 levels) in from_file'
  /opt/kitchen/cache/cookbooks/rabbitmq/recipes/users.rb:35:in `block in from_file'
  /opt/kitchen/cache/cookbooks/rabbitmq/recipes/users.rb:27:in `each'
  /opt/kitchen/cache/cookbooks/rabbitmq/recipes/users.rb:27:in `from_file'

Relevant File Content:
----------------------
/opt/kitchen/cache/cookbooks/rabbitmq/recipes/users.rb:

 31:      action :add
 32:
 33:      notifies :set_tags, "rabbitmq_user[set-tags-#{user['name']}]", :immediately
 34:    end
 35:    rabbitmq_user "set-tags-#{user['name']}" do
 36:      user user['name']
 37:      tag user['tag']
 38>>     action :set_tags
 39:    end
 40:    user['rights'].each do |r|
 41:      rabbitmq_user "set-perms-#{user['name']}-vhost-#{Array(r['vhost']).join().tr('/', '_')}" do
 42:        user user['name']
 43:        vhost r['vhost']
 44:        permissions "#{r['conf']} #{r['write']} #{r['read']}"
 45:        action :set_permissions
 46:      end
 47:    end

System Info:
------------
chef_version=15.5.17
platform=ubuntu
platform_version=16.04
ruby=ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64-linux]
program_name=/opt/chef/embedded/bin/chef-client
executable=/opt/chef/embedded/bin/chef-client
```

**Comment by michaelklishin** [comment]
I restored resource actions and tests started passing. Thank you. Would be great to have a version of this (with my patch) for `v5.x`.

**Comment by tas50** [comment]
Yep it looks like we need to keep the actions for LWRPs after all. That was the original behavior in cookstyle and I've reverted it here: https://github.com/chef/cookstyle/pull/436

**Comment by michaelklishin** [comment]
Cherry-picked to `v5.x`.

**Pull request merged**